docs/architecture/deployment-execution/import.md

```diff
@@ -0,0 +1,184 @@
+
+If all of these steps succeed, the user is left with a definition for `R` in
+their Pulumi program that matches that in the stack's state exactly.
+
+```{mermaid}
+:caption: Importing a resource using the `import` resource option
+:zoom:
+
+sequenceDiagram
+    participant LH as Language host
+    box Engine
+        participant RM as Resource monitor
+        participant SG as Step generator
+        participant SE as Step executor
+    end
+    participant P as Provider
+
+    LH->>+RM: RegisterResourceRequest(type, name, inputs, options)
+    RM->>+SG: RegisterResourceEvent(type, name, inputs, options)
+    SG->>+SE: ImportStep(inputs, options)
+    SE->>+P: ReadRequest(type, id)
+    P->>-SE: ReadResponse(current inputs, current state)
+    SE->>+P: CheckRequest(type, inputs, current inputs)
+    P->>-SE: CheckResponse(inputs', failures)
+    SE->>+P: DiffRequest(type, inputs', current state, options)
+    P->>-SE: DiffResponse(diff)
+    SE->>-RM: Done(current state)
+    RM->>-LH: RegisterResourceResponse(URN, ID, current state)
+```
+
+## `pulumi import`
+
+[`pulumi import`](https://www.pulumi.com/docs/cli/commands/pulumi_import/) is a
+newer method of importing resources into a stack that also generates program
+code for imported resources. `pulumi import` accepts a list of *import specs*,
+where each spec comprises at minimum a [type token](urns), name, and
+[ID](resource-ids), but may also specify a parent URN, provider reference, and
+package version. Unlike the `import` resource option, `pulumi import` does not
+insist that the desired state of the resource in the Pulumi program matches the
+actual state of the resource as returned by the provider, since it is capable of
+generating code to match the actual state. Given a resource `R` of type `T` with
+import ID `X` and an (initiall empty) set of input properties `Iₚ`, the engine
+performs the following sequence of operations:
+
+1. Fetch the current inputs `Iₐ` and state `Sₐ` for the resource of type `T`
+   with ID `X` from its provider by calling the provider's
+   [](pulumirpc.ResourceProvider.Read) method. If the provider does not return a
+   value for `Iₐ`, the provider does not support importing resources and the
+   import fails.
+2. Fetch the schema for resources of type `T` from the provider. If the provider
+   is not schematized or if `T` has no schema, the import fails.
+3. Copy the value of each required input property defined in the schema for `T`
+   from `Iₐ` to `Iₚ`.
+4. Validate the resource's inputs and apply any programmatic defaults by passing
+   `Iₚ` and `Iₐ` to the provider's [](pulumirpc.ResourceProvider.Check) method.
+   Let `Iₖ` be the checked inputs; these inputs form the resource's desired
+   state.
+5. Check for differences between `Iₖ` and `Sₐ` by calling the provider's
+   [](pulumirpc.ResourceProvider.Diff) method. If the provider reports any
+   differences, the values of the differing properties are copied from `Sₐ` to
```

Comment:
TIL

Reply:
I also -- a lot of this is mostly copied from what was already in the docs, with things I "know" not to be true/to be no longer true removed.